cartographer_ros/src/cartographer_node_main.cc

```diff
@@ -445,7 +493,12 @@ void Node::Initialize() {
     expected_sensor_identifiers.insert(kImuTopic);
   }
 
-  // TODO(hrapp): Add odometry subscribers here.
+  if (use_odom_) {
+    CHECK(!provide_odom_) << "Cannot use odom and provide odom at the same time";
```

Comment:
I think this isn't true. Odometry data can be published without also publishing an odom frame.